crates/trigger/src/runtime_config/key_value.rs

```diff
@@ -122,19 +124,33 @@ impl RedisKeyValueStoreOpts {
 
 #[derive(Clone, Debug, Deserialize)]
 pub struct AzureCosmosConfig {
```

Comment:
Should this struct contain an untagged enum (https://serde.rs/enum-representations.html#untagged) rather than defaulted values?

Reply:
Hmm, maybe.  It seems like a user should have `environmental_auth` _or_ `key` but not both.  But I'm not sure the two cases are different enough to justify the extra complexity and the weird unhelpful errors that untagged enums give us.  I'm actually wondering, is `environmental_auth` ever needed?  That is, is the presence of `key` enough to signal "use this key instead of the one in the environment"?  If so, would it make sense to make `key` an `Option<String>` and ditch `environmental_auth` altogether?